lib/tapioca/gem/pipeline.rb

```diff
@@ -142,7 +148,7 @@ def on_symbol(event)
         push_constant(symbol, constant) if constant
       end
 
-      sig { params(event: Gem::ConstantFound).void.checked(:never) }
+      sig { params(event: T.any(Gem::ConstantFound, Gem::NotOwnedConstantFound)).void.checked(:never) }
```

Comment:
If `NotOwnedConstantFound` subclasses `ConstantFound`, we don't need this `T.any`.